packages/dashboard/src/customization/widgets/lineSymbol/lineAnchor/component.tsx

```diff
@@ -0,0 +1,98 @@
+import { useGridSettings } from '~/components/actions/useGridSettings';
+
+/**
+ * Compute the new position of the widget based on initial position, offset difference, and grid settings.
+ *
+ * @param {XYCoord} initialPosition - The initial position of the widget.
+ * @param {XYCoord} offsetDifference - The offset difference based on the drag event.
+ * @param {number} height - The height of the widget.
+ * @param {number} width - The width of the widget.
+ * @param {number} cellSize - The size of a cell in the grid.
+ * @return {XYCoord} - The new position (x, y) for the widget.
+ */
+const computeNewPosition = (
+  initialPosition: XYCoord,
+  offsetDifference: XYCoord,
+  height: number,
+  width: number,
+  cellSize: number
+): XYCoord => {
+  const { x: initialX, y: initialY } = initialPosition;
+  const { x: offsetX, y: offsetY } = offsetDifference;
+
+  const widthPixels = Math.round(width * cellSize);
+  const heightPixels = Math.round(height * cellSize);
+
+  const newX = Math.max(0, Math.min(initialX + offsetX, widthPixels));
+  const newY = Math.max(0, Math.min(initialY + offsetY, heightPixels));
+
+  return { x: newX, y: newY };
+};
+
+/**
+ * Custom hook to handle drag-and-drop for LineWidget anchors.
+ *
+ * This hook initializes drag functionality and updates the widget position
+ * whenever a drag event occurs on either anchor.
+ *
+ * @param {LineWidget} widget - The current LineWidget object that is being manipulated.
+ * @param {'start'|'end'} anchorType - Specifies which anchor ('start' or 'end') is being dragged.
+ * @return {ConnectDragSource} - A ref to be attached to the drag component.
+ */
+const useDragAndUpdate = (
+  point: XYCoord,
+  dimensions: { height: number; width: number },
+  anchorType: 'start' | 'end',
+  updateWidget: (point: XYCoord) => void
+) => {
+  const initialPointRef = useRef({
+    x: 0,
+    y: 0,
+  });
+
+  const { cellSize } = useGridSettings();
+
+  const [{ diff, isDragging }, ref] = useDrag({
+    type: 'LineAnchor',
+    item: () => {
+      initialPointRef.current = point;
+      return {
+        anchorType,
+      };
+    },
+    collect: (monitor) => {
+      const offsetDifference = monitor.getDifferenceFromInitialOffset();
+      return { diff: offsetDifference, isDragging: monitor.isDragging() };
+    },
+  });
+
+  useEffect(() => {
+    if (isDragging && diff) {
+      const newPoint = computeNewPosition(initialPointRef.current, diff, dimensions.height, dimensions.width, cellSize);
+      updateWidget(newPoint);
+    }
+  }, [diff?.x, diff?.y]);
+  return ref;
+};
+
+export const LineAnchor: React.FC<{
+  style: CSSProperties;
+  anchorType: 'start' | 'end';
+  point: XYCoord;
+  dimensions: { height: number; width: number };
+  updateWidget: (point: XYCoord) => void;
+}> = ({ style, anchorType, point, dimensions, updateWidget }) => {
+  const dragRef = useDragAndUpdate(point, dimensions, anchorType, updateWidget);
+  return (
+    <div
+      id='line-symbol-widget'
```

Comment:
We should use class instead since this doesn't uniquely id 1 html element. There's always 2.